pkg/cli/serve/serve.go

```diff
@@ -69,23 +86,41 @@ func NewServeCmd() *cobra.Command {
 		Long:    serviceLong,
 		Example: serviceExample,
 		Args:    cobra.OnlyValidArgs,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			mode, _ := cmd.Flags().GetBool(config.FleetMode)
+			if !mode {
+				_ = cmd.MarkFlagRequired(config.AccessToken)
+				_ = cmd.MarkFlagRequired(config.ClusterID)
+			}
+			testMode, _ := cmd.Flags().GetBool(config.TestFleetMode)
```

Comment:
What it is the test fleet mode?

Reply:
Ok, I think I would need to add some documentation around it. But [comment](https://github.com/openshift/ocm-agent/pull/49#discussion_r1141676656) addresses some parts of the answer too. My thoughts were that `ocm-agent` should never consume the OIDC ID and Secret in fleetmode directly as flags but only while "testing/development" work. So I tied `ocm-agent-client-id` and `ocm-agent-client-secret` as mandatory flags only when the test mode flag is passed.

Reply:
If we just want to test fleet mode, we need the secret volume mounts on the cluster. So as an alternative to have the testing possible locally by passing strings, I added the test mode flag as well. I can simplify it by just keeping the client id and secret flags and removing the bool flag if that reduces confusion. Let me know about same.

Reply:
Have removed the testmode flag